lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -982,6 +968,41 @@ private function getParameterType(\ReflectionParameter $parameter)
         return $this->formatType($parameter->getType(), $parameter->getDeclaringFunction(), $parameter);
     }
 
+	/**
+	 * @param \ReflectionMethod $method
+	 * @param string $parameterName
+	 *
+	 * @return string|null
+	 */
+    private function getParameterTypeByMethod(\ReflectionMethod $method, string $parameterName)
+	{
+		foreach ($method->getParameters() as $parameter) {
+			if ($parameter->getName() === $parameterName) {
+				return $this->getParameterType($parameter);
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * @param \ReflectionClass $reflection
+	 * @param string $method
+	 * @param string $parameter
+	 *
+	 * @return string|null
+	 */
+	private function getMethodParameterTypeHintByReflection(\ReflectionClass $reflection, string $method, string $parameter)
+	{
+		try {
+			$typeHint = $this->getParameterTypeByMethod($reflection->getMethod($method), $parameter);
+
+			return $typeHint ? $typeHint . ' ' : null;
+		} catch (\ReflectionException $e) {
```

Comment:
This should *not* happen

Reply:
<img width="952" alt="Snímek obrazovky 2019-04-08 v 15 31 16" src="https://user-images.githubusercontent.com/4738758/55727922-5b0d7d80-5a13-11e9-89c0-33474eb58cec.png">

Can you explain me why? My IDE say I have a unhandled exception.

Thanks.

Reply:
`getMethod()` can fail if the method does not exist. Silencing the failure is not a viable solution if that happens though: if you can't handle it, re-throw instead.

Reply:
Fixed.